src/js/controllers/LottoMachineController.js

```diff
@@ -0,0 +1,111 @@
+import LottoController from "./LottoController.js";
+import LottoAnalyticsController from "./LottoAnalyticsController.js";
+import LottoMachineView from "../views/LottoMachineView.js";
+import { LOTTO, MESSAGES } from "../constants.js";
+import LottoMachine from "../models/LottoMachine.js";
+
+class LottoMachineController {
+  #lottoMachine;
+  #lottoMachineView;
+
+  #lottoController;
+  #lottoAnalyticsController;
+
+  constructor() {
+    this.#lottoController = new LottoController();
+    this.#lottoAnalyticsController = new LottoAnalyticsController();
+
+    this.#lottoMachine = new LottoMachine();
+    this.#lottoMachineView = new LottoMachineView();
+
+    this.#subscribeEvents();
+  }
+
+  #subscribeEvents() {
+    this.#lottoMachineView.$purchaseForm.addEventListener(
+      "submit",
+      this.#onPurchase.bind(this)
+    );
+    this.#lottoMachineView.$lastWinningNumbersForm.addEventListener(
+      "submit",
+      this.#onCheckWinningLottoResult.bind(this)
+    );
+    document.addEventListener("@clear", () => {
+      this.clear();
+    });
+  }
+
+  #onPurchase(event) {
+    event.preventDefault();
+
+    const purchasePrice = this.#lottoMachineView.$purchaseAmountInput.value;
+    if (this.isUnavailablePurchasePrice(+purchasePrice)) {
+      window.alert(MESSAGES.WRONG_PURCHASE_PRICE);
+      return;
+    }
+
+    if (this.isExceedPurchasePriceLimit(+purchasePrice)) {
+      window.alert(MESSAGES.WRONG_PURCHASE_PRICE);
+      return;
+    }
+
+    this.#lottoMachine.setPurchasePrice(purchasePrice);
+    const purchaseAmount = this.calculateAmountPer(purchasePrice);
+
+    this.#lottoController.onGenerateLottosBy(purchaseAmount);
+    this.#lottoMachineView.showElement(
+      this.#lottoMachineView.$lastWinningNumbersForm
+    );
+  }
+
+  #onCheckWinningLottoResult(e) {
+    e.preventDefault();
+
+    const winningNumbers = this.getWinningNumbers(e.target);
+    if (this.isWinningNumberDuplicated(winningNumbers)) {
+      window.alert(MESSAGES.WRONG_LOTTO_NUMBER);
+      return;
+    }
+
+    this.#lottoMachine.setWinningNumbers(winningNumbers);
+    this.#lottoAnalyticsController.onAnalyzeLottoResults({
+      winningNumbers: this.#lottoMachine.getWinningNumbers(),
+      lottoNumbers: this.#lottoController.getLottoNumbers(),
+      investments: this.#lottoMachine.getPurchasePrice(),
+    });
+  }
+
+  clear() {
+    this.#lottoMachine.clear();
+    this.#lottoMachineView.clear();
+  }
+
+  isWinningNumberDuplicated(numbers) {
+    const unDuplicatedNumbers = new Set();
+    numbers.forEach((number) => {
+      unDuplicatedNumbers.add(number);
+    });
+
+    return unDuplicatedNumbers.size !== LOTTO.NUMBER_COUNT_WITH_BONUS;
+  }
```

Comment:
```suggestion
  hasDuplicateElement(array) {
     return array.length === new Set(array).length
  }
```

 요렇게 표현해볼 수 있을것 같고,  유틸함수로 분리해볼 수도 있을 것 같아요.